alibi/explainers/cfproto.py

```diff
@@ -182,16 +185,15 @@ def __init__(self,
         self.clip = clip
         self.write_dir = write_dir
 
-        # compute dimensionality after conversion from OHE to ordinal encoding
-        shape = ohe_to_ord_shape(shape, cat_vars=cat_vars, is_ohe=self.ohe)
-
         if self.is_cat:
+            # compute dimensionality after conversion from OHE to ordinal encoding
+            shape = ohe_to_ord_shape(shape, cat_vars=cat_vars, is_ohe=self.ohe)
```

Comment:
I moved this inside the `if self.is_cat` block because it really doesn't make sense to do if there is no `cat_vars` (again, a case of not validating allowed combinations of kwargs).

Reply:
Seems like it might be best to open a new issue about raising a warning for incorrect combinations of kwargs?